src/net.rs

```diff
@@ -17,19 +17,123 @@
 
 // TODO: Remove this line as soon as this module is actually used
 #![allow(dead_code, unused_variables)]
+use std::net::TcpStream;
+use std::io::{Write,Read, Error, ErrorKind};
+use byteorder::{BigEndian, WriteBytesExt, ReadBytesExt}; // for write_u16()
+use std::io;
+use bincode::rustc_serialize::{decode_from, encode_into};
+use bincode::SizeLimit;
+use std::io::Cursor;
+
+#[derive(RustcEncodable, RustcDecodable)]
+#[repr(u8)]
+pub enum CNV{
+	GREETING = 0,
+	LOGIN,
+	ERROR,
+}
+#[derive(RustcEncodable, RustcDecodable)]
+#[repr(u8)]
+pub enum ErrorCode{
+	UNSPECIFIC_ERR = 0,
+	IO_ERR,
+	UNEXPEC_PKG_ERR,
+}
```

Comment:
Variants in should be written in `UpperCamelCase` in Rust (e.g. `UnspecificErr`). Enum (and type names in general) are also in `UpperCamelCase`, so please rename `CNV` to `Cnv` or ... whatever it stands for :P